builder/remote.go

```diff
@@ -65,6 +66,14 @@ func MakeRemoteContext(remoteURL string, contentTypeHandlers map[string]func(io.
 	return MakeTarSumContext(contextReader)
 }
 
+// DefaultDownloadFallbackContext creates a context with a progress reader as output.
+// This is used as default for unknown mime types, like the ones we get downloading tarballs.
+func DefaultDownloadFallbackContext(remoteURL string, writer io.Writer, size int64) func(io.ReadCloser) (io.ReadCloser, error) {
```

Comment:
I rather have this function here than inlining it in `daemon/daemonbuilder/builder.go` like we have it before. That `DetectContextFromRemoteURL` was unmanageable.